drivers/ad5592r/ad5592r.c

```diff
@@ -227,5 +229,19 @@ int32_t ad5592r_init(struct ad5592r_dev *dev)
 	if (ret < 0)
 		return ret;
 
-	return ad5592r_set_channel_modes(dev);
+	ret = ad5592r_set_channel_modes(dev);
+
+	if(init_param->int_ref == AD5592R_INT_REF_ACTIVE) {
+		if (ret < 0)
+			return ret;
```

Comment:
hmm.. shouldn't his ret check be done after `ret = ad5592r_set_channel_modes(dev);` ?
```
ret = ad5592r_set_channel_modes(dev);
if (ret < 0)
        return ret;
```

Reply:
I was thinking there is no need to check it if the internal reference is not activated because the next line would be `return ret;` either way.

Reply:
Yes but  this evaluates an extra if statement that would not be necessary if ad5592r_set_channel_modes fails.

Reply:
The concept is to exit as soon as something fails do not do extra operations.